app/services/comments/destroy.rb

```diff
@@ -0,0 +1,37 @@
+module Comments
+  class Destroy
+
+    def self.call(article_id, comment_id)
+      new(article_id, comment_id).call
+    end
+
+    def call
+      remove
+      self
+    end
+
+    attr_reader :article_id, :comment_id
+    attr_accessor :errors
+
+    private
+
+    def initialize(article_id, comment_id)
+      @article_id = article_id
+      @comment_id = comment_id
+      @errors = {}
+    end
+
+    def comment
+      @comment ||= Comment.find_by(article_id: article_id, id: comment_id)
+    end
+
+    def remove
+      if comment.present?
+        comment.destroy
+      else
+        errors["not_found"] = "Comment not found for DESTROY."
```

Comment:
More rails style way:
```
def destroy_comment
  return comment.destroy if comment.present?
  
  errors[:not_found] = I18n.t("some.internationalized.error.message")
end
```

Reply:
It's called "Guard Clause": https://anthonyharvey.hashnode.dev/guard-clauses-vs-nested-conditionals 
And it's preferred way to write conditions: much more readable.

Reply:
I18n will be in 3rd PR after tests as we discussed today :)